### PR TITLE
PE-2568:add modals

### PR DESCRIPTION
**Pull request by thiagocarvalhodev**
To see examples of the component, go to the storybook and `Modals` section.

--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/29b0vpe4dkeb8
iOS release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:ios:06ea33a95a2e0d42ffce07/releases/5q9ds0ti0nnu8

### CODE REVIEW

File: packages/ardrive_ui_library/lib/src/components/button.dart [hunk 1]
```diff
@@ -110,19 +129,26 @@ class _ArDriveButtonState extends State<ArDriveButton> {
...
-    } else {
-      width = MediaQuery.of(context).size.width * 0.8;
-    }
-    return MaterialStateProperty.resolveWith<Size>(
-      (Set<MaterialState> states) {
-        return Size(width, 56);
-      },
-    );
-  }
+  // MaterialStateProperty<Size> get _minimumSize {
```
Comment (karlprieb) [inline packages/ardrive_ui_library/lib/src/components/button.dart:132]: Can we remove this getter?

File: packages/ardrive_ui_library/lib/src/components/modal.dart [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+  final String title;
+  final String content;
+  final List<ModalAction>? actions;
+
+  @override
+  Widget build(BuildContext context) {
+    late double maxWidth;
+    final deviceWidth = MediaQuery.of(context).size.width;
+
+    if (deviceWidth < 350) {
```
Comment (karlprieb) [inline packages/ardrive_ui_library/lib/src/components/modal.dart:52]: no magic numbers please

File: packages/ardrive_ui_library/lib/src/components/button.dart (hunk 1 above)
Comment (thiagocarvalhodev) [reply packages/ardrive_ui_library/lib/src/components/button.dart]: [d52ee04](https://github.com/ardriveapp/ardrive-web/pull/799/commits/d52ee0434f74f9ddf06ad39d7100f9c6159daf1d)

File: packages/ardrive_ui_library/lib/src/components/modal.dart (hunk 2 above)
Comment (thiagocarvalhodev) [reply packages/ardrive_ui_library/lib/src/components/modal.dart]: [cf90738](https://github.com/ardriveapp/ardrive-web/pull/799/commits/cf90738408d379b897efc05f6f041c0d926c324f)

File: packages/ardrive_ui_library/lib/src/components/button.dart [hunk 3]
```diff
@@ -24,36 +30,49 @@ class _ArDriveButtonState extends State<ArDriveButton> {
...
-            shape: _shapeOutlined,
-            side: _borderSise,
-            padding: _padding,
-          ),
-          child: Text(
-            widget.text,
-            style: ArDriveTypography.headline.headline5Bold(
-              color: ArDriveTheme.of(context).themeData.colors.themeFgDefault,
+        return SizedBox(
+          height: widget.maxHeight ?? 56,
```
Comment (karlprieb) [inline packages/ardrive_ui_library/lib/src/components/button.dart:58]: can you create const for this number?

Comment (karlprieb) [reply packages/ardrive_ui_library/lib/src/components/button.dart:58]: Please, review all magic numbers

Comment (thiagocarvalhodev) [reply packages/ardrive_ui_library/lib/src/components/button.dart]: [7b8c8fc](https://github.com/ardriveapp/ardrive-web/pull/799/commits/7b8c8fc239597d56bfd4f4756a276965b2d12716)

**Review by karlprieb — APPROVED** [review]

**Pull request merged**